cmake/extensions.cmake

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# cddl_generate(<target> <cddl_file> ENTRY_TYPES <entry_types> [VERBOSE])
+#
+# Add generated code to the project for decoding CBOR.
+#
+# <cddl_file> is a file with CDDL data describing the expected CBOR data.
+#
+# <entry_types> is a list of the types defined in the CDDL file for which there
+#               should be an exposed decoding function.
+#
+# Specify VERBOSE to add printing to the generated code. Check kconfig to add
+# printing in the low level CBOR decoding code.
+#
+# The result of the function is that a library is added to the project which
+# contains the generated decoding code. The code is generated at build time using
+# a script, and generates a c file and an h file with names derived from the
+# name of the <cddl_file>.
+function(cddl_generate target cddl_file)
+  if(IS_ABSOLUTE ${cddl_file})
+    set(cddl_path ${cddl_file})
+  else()
+    set(cddl_path ${CMAKE_CURRENT_LIST_DIR}/${cddl_file})
+  endif()
+
+  if(NOT EXISTS ${cddl_path})
```

Comment:
I know I requested this myself, but I got to wonder, could a user wish to use build-time generated cddl files ?
If build time generated cddl files should be supported, then this will block such use-case.

Note, using build time generated files will most likely require changes to the custom command belox, to ensure correct build order and dependencies, so we can extend with this possibility later.

For current time, this check isgood to have.

Reply:
Yes, I was thinking the same. I agree that it's good to keep it for now since it good to have.